enable python 3.13 builds and upgrade pillow to avoid setuptools incompatibility

bosh opened the pull request:
Looks like python 13 setuptools had an issue with pillow < 10.1 or so (nested execs when establishing the __version__ info? No matter!)

There was a note that 10.x broke Kivy somehow, but no tests seem to be failing, not sure if there's something else I need to check.

bosh committed:
enable python 3.13 builds and upgrade pillow to avoid setuptools incompatibility

bosh commented:
See: https://github.com/missionpinball/mpf/pull/1915#issuecomment-3260389436

avanwinkle commented:
No tests will fail here because Kivy doesn't run with MPF, but MC depends on Kivy so upgrading that could be finnicky.

I would suggest we target these changes at 0.80 to avoid any Kivy-related shenanigans entirely, and leave 0.57 at 3.12 where MPF and Python can fade away into deprecation together.

Thanks for tackling this!

bosh commented:
Gotcha, will retarget on this branch. What would you think about dropping Python 3.8 and 3.9 with the .80 bump as well?

avanwinkle commented:
Sounds like a good idea! I'm all in favor of keeping current with versions. Thanks!

bosh committed:
drop python 3.8 and 3.9 support in prep for full release of MPF 0.80

note that these versions were not breaking with the .80 change, but have instead
just become deprecated due to the python version management schedule
